From 36948fcaf63b2808257a8709bfa67bef26251c1e Mon Sep 17 00:00:00 2001
From: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
Date: Fri, 20 Feb 2026 08:55:00 +0000
Subject: [PATCH 1/1] qsort: change defines from freebsd to cheri

This is not FreeBSD OS, so the pre-processor will not do what it should
and the postgres qsort will be used.

Lets fix it without due care.

Signed-off-by: Pawel Zalewski <pzalewski@thegoodpenguin.co.uk>
---
 src/include/port.h   | 2 +-
 src/port/qsort.c     | 2 +-
 src/port/qsort_arg.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/include/port.h b/src/include/port.h
index fe0cf89ce7..d3fc417027 100644
--- a/src/include/port.h
+++ b/src/include/port.h
@@ -495,7 +495,7 @@ extern int	pg_qsort_strcmp(const void *a, const void *b);
 typedef int (*qsort_arg_comparator) (const void *a, const void *b, void *arg);
 
 /* Use ifdef FreeBSD and not __CHERI_PURE_CAPABILITY__ so that we use the same code path for MIPS and CHERI */
-#ifdef __FreeBSD__
+#ifdef __CHERI_PURE_CAPABILITY__
 /*
  * XXXAR: the postgres version of qsort_arg does not work with capabilities (swap
  * is broken) so we have to make sure to use the libc function qsort_s instead
diff --git a/src/port/qsort.c b/src/port/qsort.c
index 8a75ff492e..50af3ebb02 100644
--- a/src/port/qsort.c
+++ b/src/port/qsort.c
@@ -46,7 +46,7 @@
 
 #include "c.h"
 
-#ifdef __FreeBSD__
+#ifdef __CHERI_PURE_CAPABILITY__
 /*
  * XXXAR: the postgres version of qsort does not work with capabilities (swap
  * is broken) so we have to make sure to use the libc version
diff --git a/src/port/qsort_arg.c b/src/port/qsort_arg.c
index 7e9b8abf54..fc7755a926 100644
--- a/src/port/qsort_arg.c
+++ b/src/port/qsort_arg.c
@@ -46,7 +46,7 @@
 
 #include "c.h"
 
-#ifndef __FreeBSD__
+#ifndef __CHERI_PURE_CAPABILITY__
 #warning "using postgres custom qsort_arg function"
 
 static char *med3(char *a, char *b, char *c,
-- 
2.43.0

